objax/functional/loss.py

```diff
@@ -48,6 +52,25 @@ def cross_entropy_logits_sparse(logits: JaxArray, labels: Union[JaxArray, int])
     return logsumexp(logits, axis=1) - logits[jn.arange(logits.shape[0]), labels]
 
 
+def mean_squared_logarithmic_error(y_true: JaxArray,
+                                    y_pred: JaxArray,
+                                    eps: float = 2 ** -17) -> JaxArray:
+    """Computes the the mean squared logarithmic error.
+
+    Args:
+        y_true: n-dimensional tensor of floats.
+        y_pred: n-dimensional tensor of floats.
+
+    Returns:
+        scalar tensor containing mean squared logarithmic error
+    """
+    y_true_log = jn.log1p(jn.maximum(y_true, eps + 1.))
+    y_pred_log = jn.log1p(jn.maximum(y_pred, eps + 1.))
+    
+    loss = (y_pred_log - y_true_log) ** 2
+    return loss.mean(axis=0)
```

Comment:
For losses, we return the vector before the mean (so that it leaves freedom for manipulation by the user. For example to use thresholding like in FixMatch. It's also better for multi-GPU. See documentation Parallelism).

So in this case, you would simply do 
```
    return (y_pred_log - y_true_log) ** 2
```

Just for my education how would it work, say I have 2 vectors y_pred, y_true of shape (2, 3, 4, 5), what does the output loss look like?

Reply:
The `mean` is always on the last axis. 
```
Input shape: (batch_size, d0, d1...dn)
Output shape: (batch_size, d0, d1...dn-1)
```

Reply:
I also noticed that I need not to put extra `1` as `log1p` takes cares of that. I will update the code. One question though: Even if we are doing `parallelism`, shouldn't we return the mean?